build/vsts_build.yaml

```diff
@@ -82,14 +82,18 @@ jobs:
 
 - job: Build_and_UnitTest
   dependsOn: Initialize_Build
-  timeoutInMinutes: 170
+  timeoutInMinutes: $[variables['timeoutlimit']]
   variables:
     BuildNumber: $[dependencies.Initialize_Build.outputs['updatebuildnumber.BuildNumber']]
     FullVstsBuildNumber: $[dependencies.Initialize_Build.outputs['updatebuildnumber.FullVstsBuildNumber']]
     VsTargetChannel: $[dependencies.Initialize_Build.outputs['updatebuildnumber.VsTargetChannel']]
     VsTargetMajorVersion: $[dependencies.Initialize_Build.outputs['updatebuildnumber.VsTargetMajorVersion']]
     SDKVersionForBuild: $[dependencies.Initialize_Build.outputs['getSDKVersionForBuild.SDKVersionForBuild']]
     LocalizedLanguageCount: "13"
+    ${{ if eq($(TimeoutInMinutesUnitTest), '') }}:  # if TimeoutInMinutesUnitTest is not defined in pipeline, use default timeoutlimit
+      timeoutlimit: 10
+    ${{ if ne($(TimeoutInMinutesUnitTest), '') }}:   # if TimeoutInMinutesUnitTest is defined in pipeline, use TimeoutInMinutesUnitTest
```

Comment:
Not sure how important it is, but 0 seems to mean "forever" in some cases. Maybe we should require > 0?